src/GCDTest.java

```diff
@@ -0,0 +1,226 @@
+        } finally {
+            posTestPassedCounter = posTestPassedCounter + 1;
+        }
+
+    }
+
+    @Test
+    public void testgcdPositive5() {
+        //first argument is a prime
+        try {
+            Assert.assertEquals(1, GCD.gcd(37, 600));
+            Assert.assertEquals("testgcdPositive5", name.getMethodName());
+        } finally {
+            posTestPassedCounter = posTestPassedCounter + 1;
+        }
+
+    }
+
+    @Test
+    public void testgcdPositive6() {
+        //one is multiplum of other
+        try {
+            Assert.assertEquals(20, GCD.gcd(20, 100));
+            Assert.assertEquals("testgcdPositive6", name.getMethodName());
+        } finally {
+            posTestPassedCounter = posTestPassedCounter + 1;
+        }
+
+    }
+
+    @Test
+    public void testgcdPositive7() {
+        //straight case
+        try {
+            Assert.assertEquals(18913, GCD.gcd(624129, 2061517));
+            Assert.assertEquals("testgcdPositive7", name.getMethodName());
+        } finally {
+            posTestPassedCounter = posTestPassedCounter + 1;
+        }
+
+    }
+
+    //Negative TestCases
+    @Test
+    public void testgcdNegative1() {
+        //Bug 1 - does not handle negative inputs
+        //Case 1 - both negative
+        try {
+            Assert.assertEquals("testgcdNegative1", name.getMethodName());
+            Assert.assertEquals(3, GCD.gcd(-6, -3));
+        } finally {
+            posTestPassedCounter = posTestPassedCounter + 1;
+        }
+
+
+    }
+    @Test
+    public void testgcdNegative2() {
+
+        //Case 2 - one negative (a will be increased forever while b does not change)
+        try {
+            Assert.assertEquals("testgcdNegative2", name.getMethodName());
+            Assert.assertEquals(4, GCD.gcd(8, -4));
+        } finally {
+            posTestPassedCounter = posTestPassedCounter + 1;
+        }
+
+
+    }
+
+    @Test
+    public void testgcdNegative3() {
+        try {
+            Assert.assertEquals("testgcdNegative3", name.getMethodName());
+            //Case 3 - one negative (b will increase forever)
+            Assert.assertEquals(4, GCD.gcd(-8, 4));
+        } finally {
+            posTestPassedCounter = posTestPassedCounter + 1;
+        }
+
+    }
+
+    @Test
+    public void testgcdNegative4() {
+        try {
+            Assert.assertEquals("testgcdNegative4", name.getMethodName());
+            // a value is zero
+            Assert.assertEquals(20, GCD.gcd(0, 20));
+        } finally {
+            posTestPassedCounter = posTestPassedCounter + 1;
+        }
+
+    }
+
+    @Test
+    public void testgcdNegative5() {
+        try {
+            Assert.assertEquals("testgcdNegative5", name.getMethodName());
+            //b value is zero
+            //should get divide by zero exception which is ArithmeticException
+            exception.expect(ArithmeticException.class);
+            int gcdResult = GCD.gcd(0, 0);
```

Comment:
We will not be fixing this in the "test code", but yes, that is definitely an error that is not handled :)